### ISSUE TITLE
Differences between paper and source

### ISSUE THREAD

**Issue by geroldmeisinger** [ivanperez-keera/dunai#319]
[Functional Reactive Programming, Refactored](http://www.cs.nott.ac.uk/~psxip1/papers/2016-HaskellSymposium-Perez-Barenz-Nilsson-FRPRefactored-short.pdf)

### 3.2 Lifting: `liftS`

in paper: `liftS :: (a -> m b) -> MSF m a b`

Searching Hoogle for `(a -> m b) -> MSF m a b` we get `arrM`. `liftS` should be introduced as an deprecated alias for `arrM`.

`{-# DEPRECATED liftS "Use arrM - the alias liftS was only used in the refactored paper." #-} liftS = arrM`

already mentioned in [issue #229](https://github.com/ivanperez-keera/dunai/issues/229#issuecomment-641852095)

### 4.1 Reader: `runReaderS`

in paper:
```
runReaderS  :: Monad m => MSF (ReaderT r m) a b -> r -> MSF m a b
runReaderS_ :: Monad m => MSF (ReaderT r m) a b -> MSF m (r,a) b
runReaderS_ :: Monad m => MSF (ReaderT s m) (s, a) b -> MSF m a b
```

in source:
```
runReaderS  :: Monad m => MSF (ReaderT r m) a b -> MSF m (r, a) b
runReaderS_ :: Monad m => MSF (ReaderT s m) a b -> s -> MSF m a b
readerS     :: Monad m => MSF m (r, a) b -> MSF (ReaderT r m) a b
```

### 4.2 Writer: `runWriterS`

in paper: `runWriterS :: Monad m => MSF (WriterT r m) a b -> MSF m a (b, r)`
in source: `runWriterS :: Monad m => MSF (WriterT r m) a b -> MSF m a (r, b)`

### 5.1 Lifting MSFs

in paper:
```
liftLM :: (Monad m, Monad n) => (forall a . m a -> n a) -> MSF m a b -> MSF n a b
liftST = liftLM . lift
```

in source:
```
-- | Lifts a monadic computation into a Stream.
constM :: Monad m => m b -> MSF m a b
constM = arrM . const

-- | Apply a monadic transformation to every element of the input stream.
-- Generalisation of 'arr' from 'Arrow' to monadic functions.
arrM :: Monad m => (a -> m b) -> MSF m a b
arrM = ...

-- | Apply trans-monadic actions (in an arbitrary way).
-- This is just a convenience function when you have a function to move across
-- monads, because the signature of 'morphGS' is a bit complex.
morphS :: (Monad m1, Monad m2)
      => (forall c . m1 c -> m2 c)
      -> MSF m1 a b
      -> MSF m2 a b
morphS = ...

-- | Lift inner monadic actions in monad stacks.
liftTransS :: (MonadTrans t, Monad m, Monad (t m))
           => MSF m a b
           -> MSF (t m) a b
liftTransS = morphS lift
```

Therefore:
```
arrM_  = constM
liftS  = arrM
liftLM = morphS
liftST = liftTransS
```

**Comment by ivanperez-keera** [comment]
I do not wish to maintain a whole list of definitions in dunai just so that the paper keeps compiling.

There is an example just to keep the paper up to date with the current API. In particular, line:

https://github.com/ivanperez-keera/dunai/blob/99299af3409b71586a6015e6dfb158c551ef2d0b/dunai-examples/paper/src/DunaiExamplesConcepts.lhs#L37

indicates that `liftS` is now `arrM` and implements the same example with `arrM`.

If you were not able to see that example, odds are that others won't either. So it makes sense to make that clearer.

I think a good way to address this is to modify the README so that, in the portion under Publications where the paper is mentioned, there is a link to the up-to-date examples from the paper.

**Comment by ivanperez-keera** [comment]
Also, some definitions are conflicting in the paper (they have more than one signature). It's great you caught that. 

We may have to list an errata somewhere. I have another one on my website.

**Comment by geroldmeisinger** [comment]
I agree. I just wanted to point the differences out somewhere. Cross-referencing all available information would be helpful I think.
